base/linalg/lapack.jl

```diff
@@ -5812,16 +5812,18 @@ for (trexc, trsen, tgsen, elty) in
             liwork = BlasInt(-1)
             info = Ref{BlasInt}()
             select = convert(Array{BlasInt}, select)
+            s = Ref{$elty}(zero($elty))
```

Comment:
The `$elty` in the `Ref` constructor takes care  of the conversion so `Ref{$elty}(0)` would be cleaner, just thought I'd mention.